src/mesh/tdycoremesh.c

```diff
@@ -489,6 +492,49 @@ PetscErrorCode SaveMeshGeometricAttributes(TDy tdy) {
     ierr = VecDestroy(&natural); CHKERRQ(ierr);
     ierr = VecDestroy(&global); CHKERRQ(ierr);
     ierr = VecDestroy(&local); CHKERRQ(ierr);
+
+    char file[PETSC_MAX_PATH_LEN];
+    PetscBool opt;
+    ierr = PetscOptionsGetString(NULL,NULL,"-tdy_connected_region",file,sizeof(file),
+                               &opt); CHKERRQ(ierr);
+    if (opt) {
+
+      PetscViewer viewer;
+      Vec region_id_nat_idx;
+
+      ierr = VecCreate(PETSC_COMM_WORLD,&region_id_nat_idx); CHKERRQ(ierr);
+
+      ierr = PetscViewerBinaryOpen(PETSC_COMM_WORLD,file,FILE_MODE_READ,&viewer); CHKERRQ(ierr);
+      ierr = VecLoad(region_id_nat_idx,viewer); CHKERRQ(ierr);
+      ierr = PetscViewerDestroy(&viewer); CHKERRQ(ierr);
+
+      // Map natural IDs in global order
+      ierr = DMCreateGlobalVector(dm, &global);CHKERRQ(ierr);
+      ierr = DMPlexNaturalToGlobalBegin(dm, region_id_nat_idx, global);CHKERRQ(ierr);
+      ierr = DMPlexNaturalToGlobalEnd(dm, region_id_nat_idx, global);CHKERRQ(ierr);
+
+      // Map natural IDs in local order
+      ierr = DMCreateLocalVector(dm, &local);CHKERRQ(ierr);
+      ierr = DMGlobalToLocalBegin(dm, global, INSERT_VALUES, local); CHKERRQ(ierr);
+      ierr = DMGlobalToLocalEnd(dm, global, INSERT_VALUES, local); CHKERRQ(ierr);
+
+      // Save the region ids
+      TDy_region *region = &mesh->region_connected;
+
+      ierr = VecGetLocalSize(local, &size_local);
+      region->num_cells = size_local/num_fields;
+      ierr = TDyAllocate_IntegerArray_1D(&region->id,region->num_cells); CHKERRQ(ierr);
+
+      ierr = VecGetArray(local, &p); CHKERRQ(ierr);
+      for (ii = 0; ii < size_local/num_fields; ++ii) region->id[ii] = p[ii*num_fields];
```

Comment:
You can use `region->num_cells` here instead of `size_local/num_fields` to make it a bit more obvious what's going on.